src/api/Feed.js

```diff
@@ -539,6 +668,171 @@ class Feed extends Base {
+            },
+            created_at: new Date().toISOString(),
+            due_at: dueAtString,
+            id: uuid,
+            name: message,
+            type: TASK,
+            assigned_to: {
+                entries: assignees.map((assignee: SelectorItem) => ({
+                    id: uniqueId(),
+                    target: { ...assignee, avatar_url: '', type: 'user' },
+                    status: TASK_NEW_INITIAL_STATUS,
+                    permissions: {
+                        can_delete: false,
+                        can_update: false,
+                    },
+                    role: 'ASSIGNEE',
+                    type: 'task_collaborator',
+                })),
+                limit: 10,
+                next_marker: null,
+            },
+            permissions: {
+                can_update: false,
+                can_delete: false,
+                can_create_task_collaborator: false,
+                can_create_task_link: false,
+            },
+            task_links: {
+                entries: [
+                    {
+                        id: uniqueId(),
+                        type: 'task_link',
+                        target: {
+                            type: 'file',
+                            ...file,
+                        },
+                        permissions: {
+                            can_delete: false,
+                            can_update: false,
+                        },
+                    },
+                ],
+                limit: 1,
+                next_marker: null,
+            },
+            status: TASK_NEW_INCOMPLETE,
+        };
+
+        this.addPendingItem(this.id, currentUser, pendingTask);
+
+        const taskPayload = { name: message, due_at: dueAtString };
+
+        this.tasksNewAPI = new TasksNewAPI(this.options);
+        this.tasksNewAPI.createTask({
+            file,
+            task: taskPayload,
+            successCallback: (taskData: Task) => {
+                this.createTaskNewSuccessCallback(file, uuid, taskData, assignees, successCallback, errorCallback);
+            },
+            errorCallback: (e: ElementsXhrError, code: string) => {
+                this.updateFeedItem(this.createFeedError(messages.taskCreateErrorMessage), uuid);
+                this.feedErrorCallback(false, e, code);
+            },
+        });
+    };
+
+    /**
+     * Callback for successful creation of a Task. Creates a task assignment
+     *
+     * @param {BoxItem} file - The file to which the task is assigned
+     * @param {string} id - ID of the feed item to update with the new task data
+     * @param {Task} task - API returned task
+     * @param {Array} assignees - List of assignees
+     * @param {Function} successCallback - the function which will be called on success
+     * @param {Function} errorCallback - the function which will be called on error     *
+     * @return {void}
+     */
+    async createTaskNewSuccessCallback(
```

Comment:
I think we've been avoiding use of async/await in anything other than the `Base` API class. It looks like this file uses promises for existing methods, so it would be good to stay consistent internally.

Reply:
It uses .then(), but nothing is calling multiple dependent promise APIs yet. It creates quite a bit of extra nesting and logic to handle data flow within the function. Is there a functional reason for avoiding async/await here?

Reply:
I see what you mean now. Looks okay as-is, then.